screens/change_pokemon/change_pokemon.lua

```diff
@@ -233,41 +233,70 @@ local function redraw_moves(self)
 		moves_count = math.max(moves_count, data.index)
 	end
 	
-	M.config[hash("change_pokemon/moves")].open.y = M.config[hash("change_pokemon/moves")].closed.y + math.ceil(moves_count/ 2) * 70
+	M.config[hash("change_pokemon/moves")].open.y = M.config[hash("change_pokemon/moves")].closed.y + moves_count * 70
 
 	for _, b in pairs(move_buttons_list) do
 		gui.delete_node(gui.get_node(b.node))
 	end
 	self.move_buttons = {}
+	self.move_buttons.delete = {}
+	self.move_buttons.pp_plus = {}
+	self.move_buttons.pp_minus = {}
 	move_buttons_list = {}
 	for i=1, moves_count do
 		local nodes = gui.clone_tree(self.move_node)
 		local txt = nodes["change_pokemon/txt_move"]
+		local pp_txt = nodes["change_pokemon/txt_move_pp"]		
 		local btn = nodes["change_pokemon/btn_move"]
 		local del = nodes["change_pokemon/btn_move_delete"]
+		local pp_plus = nodes["change_pokemon/btn_move_plus"]
+		local pp_minus = nodes["change_pokemon/btn_move_minus"]
 		local icon = nodes["change_pokemon/icon_move"]
 		gui.set_id(txt, "move_txt" .. i)
+		gui.set_id(pp_txt, "move_txt_pp" .. i)
 		gui.set_id(btn, "move_btn" .. i)
 		gui.set_id(del, "delete_move_btn" .. i)
+		gui.set_id(pp_plus, "pp_plus_move_btn" .. i)
+		gui.set_id(pp_minus, "pp_minus_move_btn" .. i)
 		gui.set_position(btn, position)
 		gui.set_enabled(btn, true)
 		gui.set_color(txt, gui_colors.HERO_TEXT_FADED)
-		table.insert(move_buttons_list, {node="move_btn" .. i, text=txt, icon=icon})
-		table.insert(self.move_buttons, {node="delete_move_btn" .. i, text=txt})
-		position.x = math.mod(i, 2) * 320
-		position.y = math.ceil((i-1)/2) * -70
+		gui.set_color(pp_txt, gui_colors.HERO_TEXT_FADED)
+		table.insert(move_buttons_list, {node="move_btn" .. i, text=txt, icon=icon, pp_text=pp_txt, pp_plus=pp_plus, pp_minus=pp_minus, delete=del})
+		table.insert(self.move_buttons.delete, {node="delete_move_btn" .. i, text=txt})
+		table.insert(self.move_buttons.pp_plus, {node="pp_plus_move_btn" .. i, text=txt})
+		table.insert(self.move_buttons.pp_minus, {node="pp_minus_move_btn" .. i, text=txt})
+
+		-- Set nodes disabled until we confirm there's actually a move there, which we will do in the below loop
+		gui.set_enabled(del, false)
```

Comment:
This is resolved! The issue was that when the button became disabled, for some reason gooey would always say we are over the button. I'm not sure if this had to do with removing and re-adding the buttons or what. Fixed by checking `a.enabled and a.over` to early return, rather than just `a.over`.